rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -107,11 +107,15 @@ NodeParameters::NodeParameters(
           rcl_yaml_node_struct_fini(params);
         });
       rclcpp::ParameterMap initial_map = rclcpp::parameter_map_from(params);
-      for (auto iter = initial_map.begin(); initial_map.end() != iter; iter++) {
-        // TODO(cottsay) implement further wildcard matching
-        if (iter->first == "/**" || iter->first == combined_name_) {
+
+      // Enforce wildcard matching precedence
+      // TODO(cottsay) implement further wildcard matching
+      const char * node_matching_names[] = {"/**", combined_name_.c_str()};
+      const size_t node_matching_names_count = sizeof(node_matching_names) / sizeof(char *);
+      for (size_t index = 0U; index < node_matching_names_count; ++index) {
+        if (initial_map.count(node_matching_names[index]) > 0) {
           // Combine parameter yaml files, overwriting values in older ones
-          for (auto & param : iter->second) {
+          for (auto & param : initial_map.at(node_matching_names[index])) {
```

Comment:
Definitely +1 to the idea, but can we use more modern C++ stuff here?  Something like:

```
  std::vector<std::string> node_matching_names{"/**", combined_name};
  for (const std::string & name : node_matching_names) {
    if (initial_map.count(name.c_str()) > 0) {
      for (auto & param : initial_map.at(name.c_str())) {
      }
    }
  }
```

(alternatively, if you don't want to do `name.c_str()` all of the time, you could make that a `std::vector` of `const char *` instead)

Reply:
I made the `rclpy` patch and the C vibe stuck with me :sweat_smile: 

Changed in https://github.com/ros2/rclcpp/pull/1094/commits/e9ba4f2715cac5867f053bda857057e4e2c81e13.

Reply:
> I made the `rclpy` patch first and the C vibe stuck with me sweat_smile

Ha, yeah, totally understood.